src/value.h

```diff
@@ -23,24 +23,28 @@ typedef char *SIId;
 typedef enum {
   T_NULL = 0,
   T_STRING = 0x001,
-  T_INT32 = 0x002,
-  T_INT64 = 0x004,
-  T_UINT = 0x008,
-  T_BOOL = 0x010,
-  T_FLOAT = 0x020,
-  T_DOUBLE = 0x040,
-  T_PTR = 0x080,
+  T_CONSTSTRING = 0x002,
```

Comment:
Backward compatibility break,
we could have introduce this type as 0x400, or 0x200 and shift only T_INF and T_NEGINF as I don't believe these are being used.

Reply:
Ah, I hadn't thought of that. We'll have to change the logic for comparing disjoint types a bit, as it relies on string types being lower than numerics, but that's no big deal.

Reply:
Never mind, that comparability issue was already resolved!